nni/retiarii/oneshot/pytorch/base_lightning.py

```diff
@@ -58,7 +58,7 @@ def traverse_and_mutate_submodules(
     module_list = []
 
     def apply(m):
-        for name, child in m.named_children():
+        for name, child in list(m.named_children()):
```

Comment:
why?

Reply:
oh it's a generator and this loop body may affect the generator's behavior?
very fancy

Reply:
Yes. setattr might replace some of the "children" in-place.
The original version doesn't work at all.